### PR TITLE
[Doc] update datacache populate doc

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## Why I'm doing:
Change doc

## What I'm doing:
Change doc

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

## Documentation PRs only:

If you are submitting a PR that adds or changes English documentation and have not
included Chinese documentation, then you can check the box to request GPT to translate the
English doc to Chinese. Please ensure to uncheck the **Do not translate** box if translation is needed.
The workflow will generate a new PR with the Chinese translation after this PR is merged.

- [ ] Yes, translate English markdown files with GPT
- [x] Do not translate

### CODE REVIEW

File: docs/zh/data_source/data_cache.md [hunk 1]
```diff
@@ -73,6 +67,36 @@ SET GLOBAL enable_scan_datacache=false;
 
 ## 填充 Data Cache
 
+### 填充规则
+
+自 v3.3.2 起，为了提高 Data Cache 的缓存命中率，先按照如下默认规则填充：
+
+* 对于非 SELECT 的查询， 不进行填充。比如 `ANALYZE TABLE`，`INSERT INTO xxx SELECT xx` 等等。
+
+* 对一个表分区全扫的时候（只有一个分区则不考虑），不进行填充。
```
Comment (GavinMar) [inline docs/zh/data_source/data_cache.md:76]: "只有一个分区则不考虑"放在这里有些歧义。

File: docs/zh/data_source/data_cache.md [hunk 2]
```diff
@@ -59,7 +53,7 @@ StarRocks 以 BE 节点的内存和磁盘作为缓存的存储介质，支持全
 
 - 系统变量 `enable_scan_datacache` 和 BE 参数 `datacache_enable` 默认设置为 `true`。
 - 如未手动配置缓存路径和内存以及磁盘上限，系统会自动选择相应的路径并设置上限：
-  - 在 `storage_root_path` 的父目录下创建 **datacache** 目录作为磁盘缓存目录。（您可以通过 BE 参数 `datacache_disk_path` 修改。）
+  - 在 `storage_root_path` 的目录下创建 **datacache** 目录作为磁盘缓存目录。（您可以通过 BE 参数 `datacache_disk_path` 修改。）
```
Comment (GavinMar) [inline docs/zh/data_source/data_cache.md:56]: 去掉"的"，改成 "在`storage_root_path` 目录下创建"

File: docs/zh/data_source/data_cache_observe.md [hunk 3]
```diff
@@ -74,9 +74,50 @@ mysql> select * from information_schema.be_datacache_metrics;
...
 
-通过访问以下 API 接口，可获取该节点 Data Cache 的详细指标。
+* `/api/datacache/app_stat`：绝大多数用户关心这个指标就够了，它能最直观的反应当前缓存命中率的状态。
+* `/api/datacache/stat`：对于资深用户，可以通过此 api 得到底层 Data Cache 的执行状态。
+
+两者的区别？
+
+Data Cache 作为 StarRocks 的一个底层模块，因 StarRocks 执行层面的优化，会导致 `/api/datacache/app_stat` 和 `/api/datacache/stat` 观测的结果并不相同。
+
+比如 StarRocks 对一个没有读取过的 block 反复读取了 5 次，站在 StarRocks 的角度（`/api/datacache/app_stat`），它的缓存命中率是 0%。而对于 Data Cache 模块（`/api/datacache/stat`），它的命中率是 4/5 = 80%。
```
Comment (GavinMar) [inline docs/zh/data_source/data_cache_observe.md:86]: 这个缓存命中率是0%，站在StarRocks的角度也不好理解啊

File: docs/zh/reference/System_variable.md [hunk 4]
```diff
@@ -501,11 +501,11 @@ SELECT /*+ SET_VAR
 * 默认值：false
 * 引入版本：v2.5
 
-### enable_populate_datacache（2.5 及以后）
+### populate_datacache_mode
 
-* 描述：StarRocks 从外部存储系统读取数据时，是否将数据进行缓存。默认值为 `true`。如果只想读取，不进行缓存，可以将该参数设置为 `false`。在 3.2 之前各版本中，对应变量为 `enable_populate_block_cache`。
-* 默认值：true
-* 引入版本：v2.5
+* 描述：StarRocks 从外部存储系统读取数据时，管理数据缓存的规则。默认值为 `auto`，即自动的根据查询的特点，选择性的进行缓存。如果总是想缓存，设置为 `always` 即可。如果永远不想缓存，设置为 `never` 即可。
```
Comment (GavinMar) [inline docs/zh/reference/System_variable.md:506]: "管理数据缓存的规则"->"控制数据缓存填充行为"

File: docs/zh/data_source/data_cache_observe.md (hunk 3 above)
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache_observe.md:86]: 我也很蛋疼，想不出来好的解释说法

Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache_observe.md:86]: 来点意见

File: docs/zh/data_source/data_cache.md (hunk 1 above)
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache.md:76]: 那就删了吧？

Comment (GavinMar) [reply docs/zh/data_source/data_cache.md:76]: > 那就删了吧？

我觉得可以换个说法说明下，比如：对一个表分区全扫的时候，不进行填充。但如果只有一个分区则例外，默认进行填充。

Comment (GavinMar) [reply docs/zh/data_source/data_cache.md:76]: 直接删掉的话可能后面有有些用户觉得奇奇怪怪的现象让你分析了

File: docs/zh/data_source/data_cache_observe.md (hunk 3 above)
Comment (GavinMar) [reply docs/zh/data_source/data_cache_observe.md:86]: 不行就换个好理解的例子，比如预读这种。用户读100K，我们预读了256K，然后填充的缓存中，接下来再读100K后面紧邻的100K时直接从cache中读取了，这个时候cache命中率显示是50%，但是上层其实一共读了256K。大概就这种可能比较好解释点

File: docs/zh/data_source/data_cache.md (hunk 2 above)
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache.md:56]: done

File: docs/zh/data_source/data_cache.md (hunk 1 above)
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache.md:76]: done

File: docs/zh/data_source/data_cache_observe.md (hunk 3 above)
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache_observe.md]: changed

File: docs/zh/reference/System_variable.md (hunk 4 above)
Comment (Smith-Cruise) [reply docs/zh/reference/System_variable.md]: done

**Review by GavinMar — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Pull request merged**